ocaml/parsing/pprintast.ml

```diff
@@ -355,10 +365,13 @@ let maybe_modes_type pty ctxt f c =
   | None -> pty ctxt f c
 
 let modalities_type pty ctxt f pca =
-  match pca.pca_modalities with
+  let modalities = Split_modalities.split_list pca.pca_modalities in
+  match modalities.legacy with
   | [] -> pty ctxt f pca.pca_type
```

Comment:
I think this will fail to print the nonlegacy modalities if there are no legacy modalities

Reply:
Yep you're right

Reply:
Fixed